server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java

```diff
@@ -0,0 +1,413 @@
+                            LinkedHashMap::new, Collectors.toList()));
+                    return globalTransactionDOs.stream()
+                        .map(globalTransactionDO -> getGlobalSession(globalTransactionDO,
+                            branchTransactionDOsMap.get(globalTransactionDO.getXid())))
+                        .collect(Collectors.toList());
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public List<GlobalSession> readSession(SessionCondition sessionCondition) {
+        try (Jedis jedis = JedisPooledFactory.getJedisInstance()) {
+            if (!StringUtils.isEmpty(sessionCondition.getXid())) {
+                String globalSessionJson = jedis.get(getGlobalKeyByXid(sessionCondition.getXid()));
+                if (!StringUtils.isEmpty(globalSessionJson)) {
+                    GlobalSession session =
+                        convertGlobalSession(JSON.parseObject(globalSessionJson, GlobalTransactionDO.class));
+                    List<GlobalSession> globalSessions = new ArrayList<>();
+                    globalSessions.add(session);
+                    return globalSessions;
+                }
+            } else if (sessionCondition.getTransactionId() != null) {
+                String global = jedis.get(getGlobalKeyByTransactionId(sessionCondition.getTransactionId()));
+                if (StringUtils.isEmpty(global)) {
+                    return null;
+                }
+                GlobalTransactionDO globalTransactionDO = JSON.parseObject(global, GlobalTransactionDO.class);
+                String branchKey = getBranchListKeyByXid(globalTransactionDO.getXid());
+                Set<String> keys = lRange(jedis, branchKey);
+                List<BranchTransactionDO> branchTransactionDOs;
+                if (CollectionUtils.isNotEmpty(keys)) {
+                    branchTransactionDOs = getBranchJsons(jedis, keys);
+                    GlobalSession globalSession = getGlobalSession(globalTransactionDO, branchTransactionDOs);
+                    if (globalSession != null) {
+                        List<GlobalSession> globalSessions = new ArrayList<>();
+                        globalSessions.add(globalSession);
+                        return globalSessions;
+                    }
+                }
+            } else if (CollectionUtils.isNotEmpty(sessionCondition.getStatuses())) {
+                return readSession(sessionCondition.getStatuses());
+            }
+        }
+        return null;
+    }
+
+    private List<BranchTransactionDO> getBranchJsons(Jedis jedis, Set<String> keys) {
+        List<BranchTransactionDO> branchTransactionDOs = new ArrayList<>();
+        List<String> branchJsons = jedis.mget(keys.toArray(new String[0]));
+        for (String branchJson : branchJsons) {
+            if (!StringUtils.isEmpty(branchJson)) {
+                branchTransactionDOs.add(JSON.parseObject(branchJson, BranchTransactionDO.class));
```

Comment:
The object converted by JSON needs to be judged to be empty

Reply:
> The object converted by JSON needs to be judged to be empty

branchJson not empty,that should be enough